### PR TITLE
Minor speed-up for sum sq dev when jit compiled (slower if not)

### PR DESCRIPTION
**Pull request by rjenc29**
Morning - what do you reckon on this?  It's a little faster when jit compiled and skips the abs step since the diff is squared anyway.  (But it's slower if not jit compiled.)  Just a thought.

### CODE REVIEW

**Comment by dwillmer** [comment]
Awesome, thanks!

Definitely worth having a discussion about this - if it's faster jitted then we should include it.

My plan for the project is to have multiple versions of functions depending on accuracy/performance characteristics, so if you've sped it up that's great and we should merge.

Could you add a notebook/markdown file showing the performance differences over a range of data sizes, say 10-1e8?

i have a couple of comments on the code as well, but nothing major,

Thanks!

File: fastats/maths/sum_sq_dev.py [hunk 1]
```diff
@@ -15,7 +14,13 @@ def sum_sq_dev(x):
     >>> sum_sq_dev(x) # doctest: +ELLIPSIS
     25681.60000...
     """
-    return sum(power(abs(x - mean(x)), 2))
+    mean = x.mean()
```
Comment (dwillmer) [inline fastats/maths/sum_sq_dev.py]: The idea is that the ast transform in fastats will optionally change standalone functions to be jit versions, so we should aim for `mean(x)` over `x.mean()`, as that then allows us to choose a faster/better version later.

File: fastats/maths/sum_sq_dev.py [hunk 2]
```diff
@@ -15,7 +14,13 @@ def sum_sq_dev(x):
...
+    total = 0.0
+
+    for value in x:
+        total += (value - mean) ** 2
```
Comment (dwillmer) [inline fastats/maths/sum_sq_dev.py]: Removing the `abs` is great, thanks!

**Comment by rjenc29** [comment]
Top stuff, yep - let's chew it over - I have a few questions jotted down & will follow up next week.  Loving the OLS stuff btw :)

**Comment by rjenc29** [comment]
Hey, so I just took out the abs from the original fn.  Will create branch my fork and put in the possible optimised version with some evidence on speed-up and ping you & some other ideas

**Comment by rjenc29** [comment]
I added a notebook to one of my branches showing evidence of the possible speedup - lemme know what you think: 
https://github.com/rjenc29/fastats/blob/projects/fastats/notebooks/Experimental%20ideas.ipynb

**Comment by dwillmer** [comment]
Perfect, that notebook says it all, thanks!

Perhaps stick the notebook as 'sum_sq_dev' in a `_notebooks` folder under fastats.maths ?

Alternatively we could create a top level notebooks folder with the same hierarchy as the code.

**Comment by dwillmer** [comment]
@rjenc29 do you want to add the notebook to this commit and we can merge this one as well?

**Pull request closed without merging**